spec/services/deals_service_spec.rb

```diff
@@ -59,6 +60,22 @@
     end
   end
 
+  describe :upsert do
+    it 'raises a TypeError if filters is nil' do
+      expect(client.deals.upsert(nil, { name: 'unique_name' }).to raise_error(TypeError)
+    end
+
+    it 'raises an ArgumentError if filters is empty' do
+      expect(client.deals.upsert({}, { name: 'unique_name' }).to raise_error(ArgumentError)
+    end
+
+    it 'calls the upsert route with encoded filters' do
+      filters = { name: 'unique_name', 'custom_fields[external_id]': 'unique-1' }
+      attributes = filters.merge('custom_fields[category]': 'bags')
+      expect(client.deals.upsert(filters, attributes)).to be_instance_of BaseCRM::Deal
```

Comment:
these tests seems to require an Access Token and actually attempt to hit a running API, is that expected?